BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs

```diff
@@ -239,19 +239,12 @@ public void ClearSceneAnchors()
             m_anchors.Clear();
         }
 
-        private void UnpersistAnchor(ARAnchor anchor)
+        private void ChangeAnchorVisual(ARAnchor anchor, string newName, bool isPersisted)
         {
-            if (m_anchorStore == null)
-            {
-                Debug.Log($"Anchor Store was not available.");
-                return;
-            }
-
             SampleAnchor sampleAnchor = anchor.GetComponent<SampleAnchor>();
-            m_anchorStore.UnpersistAnchor(sampleAnchor.Name);
-            Debug.Log($"Anchor forgotten: {anchor.trackableId}");
-            sampleAnchor.Name = "";
-            sampleAnchor.Persisted = false;
+            Debug.Log(isPersisted ? $"Anchor persisted: {anchor.trackableId}" : $"Anchor unpersisted: {anchor.trackableId}");
```

Comment:
you can actually use sampleAnchor.Persisted to get the persisted state. It need not be passed in as an argument here

Reply:
isPersisted is the new state, not the current state. Hence it needs to be passed right?

Reply:
Not sure, when you clear anchor store the state of the anchors should technically be updated. It seems odd to pass in the state this way

Reply:
The immediate effect that we want to see (anchors changing color) is governed by the "persisted" value that is part of the test logic.